### PR TITLE
Reorder nodes from ``create_meshgrid`` to match other sorting

### PR DESCRIPTION
**Pull request by f0uriest**
- Ensure all tensor product grids are flattened in a consistent order
- adds a utility method for reshaping flattened data into ndarrays such that each coordinate varies along a single axis

### CODE REVIEW

File: tests/test_grid.py [hunk 1]
```diff
@@ -738,20 +738,61 @@ def test_meshgrid(self):
...
+        np.testing.assert_allclose(
+            grid.meshgrid_reshape(grid.nodes[:, 1], "rzt")[:, :, 3], t[3]
+        )
+        # reshaping tzr should have zeta along 2nd axis
+        np.testing.assert_allclose(
+            grid.meshgrid_reshape(grid.nodes, "tzr")[:, 0, :, 2], z[0]
+        )
+        np.testing.assert_allclose(
+            grid.meshgrid_reshape(grid.nodes, "tzr")[:, 3, :, 2], z[3]
+        )
```
Comment (ddudt) [inline tests/test_grid.py:786]: Why is the array 4D in this case? And is there a reason why we can't check all the nodes instead of only indices 0 and 3?

Comment (f0uriest) [reply tests/test_grid.py:786]: we're reshaping the full nodes array from `(K,3)` to `(Nt, Nz, Nr, 3)`, in the other cases we're only reshaping a single column from `(K,)` to `(Nr, Nt, Nz)` etc.

**Review by unalmis — COMMENTED** [review]
Looks good but I have a request to move a test here.

If I remember correctly, in that test the reason I'm using a function for reshaping rather than coordinates themselves is because the functions are not generated by a tensor-product while the coordinates are. This might be better because the values of the coordinates may get mixed up but not be detected because they have the same value.

**Review by unalmis — DISMISSED** [review]

File: desc/grid.py [hunk 2]
```diff
@@ -598,6 +598,52 @@ def replace_at_axis(self, x, y, copy=False, **kwargs):
...
+            f"order should be a permutation of {self.coordinates}, got {order}",
+        )
+        shape = (self.num_poloidal, self.num_rho, self.num_zeta)
+        vec = False
+        if x.ndim > 1:
+            vec = True
+            shape += (-1,)
+        x = x.reshape(shape, order="F")
+        x = jnp.moveaxis(x, 1, 0)  # now shape rtz/raz etc
+        newax = tuple(list(self.coordinates).index(c) for c in order)
```
Comment (unalmis) [inline desc/grid.py:641]: don't need call to list

File: desc/grid.py [hunk 3]
```diff
@@ -598,6 +598,52 @@ def replace_at_axis(self, x, y, copy=False, **kwargs):
...
+        if vec:
+            newax += (3,)
```
Comment (unalmis) [inline desc/grid.py:643]: -1 would be better here because I sometimes I am reshaping a subset of the 3 components of a vector.
For example to perform this operation with `meshgrid_reshape`, we need to use `-1` instead of `3`
``grid.nodes[:, 1:].reshape(num rho, -1, 2)```

Comment (unalmis) [reply desc/grid.py:643]: Actually  never mind, this is just transpose order

**Review by unalmis — APPROVED** [review]

**Review by dpanici — APPROVED** [review]

File: desc/grid.py [hunk 4]
```diff
@@ -598,6 +598,52 @@ def replace_at_axis(self, x, y, copy=False, **kwargs):
...
+            sorted(order) != sorted(self.coordinates),
+            ValueError,
+            f"order should be a permutation of {self.coordinates}, got {order}",
+        )
+        shape = (self.num_poloidal, self.num_rho, self.num_zeta)
+        vec = False
+        if x.ndim > 1:
+            vec = True
+            shape += (-1,)
+        x = x.reshape(shape, order="F")
```
Comment (unalmis) [inline desc/grid.py:639]: `f = grid.meshgrid_reshape(f)` seems simpler than `f = f.reshape(grid.num_poloidal, grid.num_radial, grid.num_toroidal, order="F").`

Is it worth to default `order=None` and early exit?

```python
inorder = self.coordinates[1] + self.coordinates[0] + self.coordinates[2]
order = setdefault(order, inorder)
...
x = x.reshape(shape, order="F")
if order == inorder:
  return x
x = jnp.swapaxes(x, 1, 0)
```

Comment (f0uriest) [reply desc/grid.py:639]: I considered that but thought it would be clearer and less error prone to make the ordering explicit.

**Comment by rahulgaur104** [comment]
Created and merged in 8 hours is wild!
